service/worker/archiver/history_blob_iterator.go

```diff
@@ -104,14 +113,15 @@ func (i *historyBlobIterator) Next() (*HistoryBlob, error) {
 	if !i.HasNext() {
 		return nil, errIteratorDepleted
 	}
-	events, nextPersistencePageToken, historyEndReached, err := i.readBlobEvents(i.persistencePageToken)
+	events, nextPersistencePageToken, historyEndReached, numEventsToSkip, err := i.readBlobEvents(i.persistencePageToken, i.numEventsToSkip)
```

Comment:
This method is returning a lot of things now, what do you think about making a struct for the result?

Reply:
In my draft pr for recording upload progess using heartbeating, the state of the iterator will be a struct and I can make this function return that struct as part of its result. So how about I making the change in that diff?

Reply:
Sounds good